### DISCUSSION TITLE
Custom activation for Saliency Maps

### DISCUSSION THREAD

**Discussion by trinhdhk** [Project-MONAI/MONAI#6012] [General]
I noticed that all gradient-based activation map methods implemented in MONAI implicitly assume ReLU as the target activation layer. This may very well be not the case (as in my case, it is SiLU). This leads to unreliable behaviour. Is it possible and sensible to write a custom autograd activation layer for visualising the activation map?

Thanks.

**Comment by KumoLiu** [comment]
Hi @trinhdhk, do you mean the target layer you want to generate the feature map? I think you could specify the `target_layers`. Here are some examples.
https://github.com/Project-MONAI/MONAI/blob/11745a6f7683effd40ff55f88306d4463f9d66a9/monai/visualize/class_activation_maps.py#L233-L253

Hope it can help you, thanks!

**Reply by trinhdhk** [reply]
Thanks. No, not exactly the GradCAM. For example, here is the source code of GuidedSm
```python
class GuidedBackpropGrad(VanillaGrad):
   

    def __call__(self, x: torch.Tensor, index: torch.Tensor | int | None = None, **kwargs) -> torch.Tensor:
        with replace_modules_temp(self.model, "relu", _GradReLU(), strict_match=False):
            return super().__call__(x, index, **kwargs)
```

I think the method tries to look for "relu" layers and replace it with a grad-hooked customised one. This assumingly might not work with other activation functions. Is there a way to implement a grad-hooked version of SiLU for instance?

**Reply by wyli** [reply]
thanks @trinhdhk, I'm creating a feature request from your suggestion.
